implementations/rust/ockam/ockam_command/src/node/mod.rs

```diff
@@ -95,3 +95,21 @@ pub fn default_node_name() -> String {
         .map(|n| n.config.name)
         .unwrap_or_else(|_| "default".to_string())
 }
+
+pub fn node_name_parser(node_name: &str) -> Result<String> {
+    if node_name == "default" && CliState::try_default().unwrap().nodes.default().is_err() {
+        return Ok(spawn_default_node(node_name));
+    }
+
+    Ok(node_name.to_string())
+}
+
+pub fn spawn_default_node(node_name: &str) -> String {
+    let config = OckamConfig::load().expect("Failed to load config");
+    let global_opts = CommandGlobalOpts::new(GlobalArgs::default(), config);
+    let mut create_command = CreateCommand::default();
```

Comment:
I don't know why I thought this didn't work with clap, but I'm glad it does. This will help us a lot in our quest to simplify how we compose commands.